primitives/linux-port/column.cpp

```diff
@@ -1420,67 +1576,181 @@ void vectorizedFiltering(NewColRequestHeader* in, ColResultHeader* out, const T*
     }
   }
 
-  // main loop
-  // writeMask tells which values must get into the result. Includes values that matches filters. Can have
-  // NULLs. nonEmptyMask tells which vector coords are not EMPTY magics. nonNullMask tells which vector coords
-  // are not NULL magics.
-  for (uint16_t i = 0; i < iterNumber; ++i)
+  if (hasAuxCol)
```

Comment:
Could you elaborate on why did you decide to make a very close copy of the main loop for the scope when hasAuxCol is true?